src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.ParseMapModules.cs

```diff
@@ -0,0 +1,125 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+internal static partial class Interop
+{
+    internal static partial class procfs
+    {
+        private const string MapsFileName = "/maps";
+
+        private static string GetMapsFilePathForProcess(int pid)
+        {
+            return RootPath + pid.ToString(CultureInfo.InvariantCulture) + MapsFileName;
+        }
+
+        internal static ProcessModuleCollection? ParseMapsModules(int pid)
+        {
+            try
+            {
+                return ParseMapsModulesCore(File.ReadLines(GetMapsFilePathForProcess(pid)));
+            }
+            catch (IOException) { }
+            catch (UnauthorizedAccessException) { }
+
+            return null;
+        }
+
+        private static ProcessModuleCollection ParseMapsModulesCore(IEnumerable<string> lines)
+        {
+            Debug.Assert(lines != null);
+
+            ProcessModule? previous = null;
+            var modules = new ProcessModuleCollection(0);
+
+            // Parse each line from the maps file into a ProcessModuleCollection result
+            foreach (string line in lines)
+            {
```

Comment:
Nit: the iteration body seems large to me. Consider either extracting the parse lambdas into local functions and/or extracting the line parser into a standalone method.

Reply:
Extracted lambdas in static methods. Also I just realized that `new ProcessModuleCollection(0)` is same as `new ProcessModuleCollection()`: https://github.com/dotnet/runtime/blob/301056d6ff8f5ac0dea7fe07a8b450754ad846cb/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessModuleCollection.cs#L19-L25